File: Offer06/answer.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def reversePrint(self, head):
        """
        :type head: ListNode
        :rtype: List[int]
        """
        result = []
        node = head
        while node != None:
            result.append(node.val)
            node = node.next
        
        return result[::-1]

File: 203/answer.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        ListNode *currentNode, *lastNode;
        currentNode = head;
        while (currentNode != NULL){
            if (currentNode == head){
                currentNode = currentNode ->next;
                lastNode = head;
            }else{
                if (currentNode -> val == val){
                    lastNode->next = currentNode ->next;
                    currentNode = lastNode->next;
                }else{
                    lastNode = currentNode;
                    currentNode = currentNode ->next;
                }
            }
        }

        if (!head){
            return NULL;
        }

        if (head->val == val){
            return head->next;
        }

        return head;

    }
};
